psi4/driver/procrouting/proc.py

```diff
@@ -1227,7 +1227,7 @@ def scf_helper(name, post_scf=True, **kwargs):
         core.set_legacy_wavefunction(ref_wfn)
 
         # Compute dftd3
-        if "_disp_functor" in dir(ref_wfn):
+        if "_disp_functor" in dir(scf_wfn) and not ref_wfn._disp_functor.excludelist:
             disp_energy = ref_wfn._disp_functor.compute_energy(ref_wfn.molecule())
             ref_wfn.set_variable("-D Energy", disp_energy)
         ref_wfn.compute_energy()
```

Comment:
I still think this is the wrong place to do these checks. We're printing the "Empirical Dispersion Energy =" for all DFT's anyway, as it seems to be always printed; otherwise it's always ignored:

https://github.com/psi4/psi4/blob/a7fc0506050facb3b544cc4200f708a5dd63c56a/psi4/src/psi4/libscf_solver/hf.cc#L1825-L1829

The EmpiricalDispersion class can, IMO, return 0 for `-NL` safely - unless we want to be ready for when somebody writes a method using both -D3 and -NL at the same time!

The more exceptions we add here, the more often we'll have to touch `proc.py`. I'm sure '-NL' is not the last - I believe `-D4` will need its own handling, as it calculates partial charges on the fly.

Reply:
Short-range NL with long-range D3, why not? I am indifferent to the solution. I thought this is smoother and the exclude list can easily be expanded.

I assume -D4 will all be handled by an external program, since gfn-xtb data is recommended as default ingredient iirc.

Reply:
Seems I am breaking the psi4-side D2 detection with it as well.

Reply:
Part of your tests fail because you're using `scf_wfn` instead of `ref_wfn`.

Reply:
i know, but my excludelist function does not work for "p4". Getting too late today, though.